### PR TITLE
Improvе "center" watcher  "ui-gmap-circle" directive for "geoJSON" compatibility

### PR DESCRIPTION
**Pull request by AbstractVisions**
Add "coordinates" reference checking and object reference checking of "geoJSON" to $watchCollection for "ui-gmap-circle" directive
https://github.com/angular-ui/angular-google-maps/issues/1679

### CODE REVIEW

File: src/coffee/directives/api/models/parent/circle-parent-model.coffee [hunk 1]
```diff
@@ -26,7 +26,7 @@ angular.module('uiGmapgoogle-maps.directives.api.models.parent')
 
       @setMyOptions = (newVals, oldVals) =>
         return if scope.settingFromDirective
-        unless _.isEqual newVals,oldVals
+        unless _.isEqual newVals,oldVals and newVals == oldVals and (newVals? and oldVals? and newVals.coordinates == oldVals.coordinates)
```
Comment (nmccready) [inline src/coffee/directives/api/models/parent/circle-parent-model.coffee]: To be safe here `_.isEqual newVals,oldVal` should have `()` as the parser somtimes dudges this `_.isEqual(newVals,oldVal)`.

Comment (nmccready) [inline src/coffee/directives/api/models/parent/circle-parent-model.coffee]: Please run grunt to make sure the specs pass locally. But you can revert the dist changes.

Comment (AbstractVisions) [inline src/coffee/directives/api/models/parent/circle-parent-model.coffee]: Yes, you are right. I will fix it!

**Comment by nmccready** [comment]
Lgtm

**Pull request merged**